src/WCSimPhysicsList.cc

```diff
@@ -392,349 +369,341 @@ void WCSimPhysicsList::ConstructGeneral()
+
+// Include the FTF model - FDL
+#include "G4FTFModel.hh"
+#include "G4LundStringFragmentation.hh"
+#include "G4ExcitedStringDecay.hh"
+#include "G4PreCompoundModel.hh"
+#include "G4GeneratorPrecompoundInterface.hh"
+#include "G4TheoFSGenerator.hh"
 
 void WCSimPhysicsList::ConstructHad() 
 {
 
-// Makes discrete physics processes for the hadrons, at present limited
-// to those particles with GHEISHA interactions (INTRC > 0).
-// The processes are: Elastic scattering and Inelastic scattering.
-// F.W.Jones  09-JUL-1998
-// 
 // This code stolen from:
 // examples/advanced/underground_physics/src/DMXPhysicsList.cc
 // CWW 2/23/05
 //
-  
+
   G4HadronElasticProcess* theElasticProcess = new G4HadronElasticProcess;
-  G4LElastic* theElasticModel = new G4LElastic;
+  G4HadronElastic* theElasticModel = new G4HadronElastic;
   theElasticProcess->RegisterMe(theElasticModel);
-  
-  theParticleIterator->reset();
-  while ((*theParticleIterator)()) 
+
+  aParticleIterator->reset();
+  while ((*aParticleIterator)()) 
     {
-      G4ParticleDefinition* particle = theParticleIterator->value();
+
+     // Add the FRITIOF model - FDL
+      G4TheoFSGenerator* FTFP_model = new G4TheoFSGenerator();
+      G4GeneratorPrecompoundInterface* theCascade = new G4GeneratorPrecompoundInterface();
+      G4ExcitationHandler* theHandler = new G4ExcitationHandler();
+      G4PreCompoundModel* thePreEquilib = new G4PreCompoundModel(theHandler);
+      theCascade->SetDeExcitation(thePreEquilib);
+      FTFP_model->SetTransport(theCascade);
+      G4LundStringFragmentation* theFragmentation = new G4LundStringFragmentation();
+      G4ExcitedStringDecay* theStringDecay = new G4ExcitedStringDecay(theFragmentation);    
+      G4FTFModel* theStringModel = new G4FTFModel;
+      theStringModel->SetFragmentationModel(theStringDecay);
+      FTFP_model->SetHighEnergyGenerator(theStringModel);
+ 
+      G4ParticleDefinition* particle = aParticleIterator->value();
       G4ProcessManager* pmanager = particle->GetProcessManager();
       G4String particleName = particle->GetParticleName();
 
-      if (particleName == "pi+") 
+      if (particleName == "pi+")
 	{
 	  pmanager->AddDiscreteProcess(theElasticProcess);
 	  G4PionPlusInelasticProcess* theInelasticProcess = 
 	    new G4PionPlusInelasticProcess();
-	  G4LEPionPlusInelastic* theLEInelasticModel = 
-	    new G4LEPionPlusInelastic;
-	  theInelasticProcess->RegisterMe(theLEInelasticModel);
-	  G4HEPionPlusInelastic* theHEInelasticModel = 
-	    new G4HEPionPlusInelastic;
-	  theInelasticProcess->RegisterMe(theHEInelasticModel);
+
+	  // Added Bertini Model - FDL
+	  G4CascadeInterface* theBertiniModel = new G4CascadeInterface;
+	  theBertiniModel->SetMaxEnergy(5.0*CLHEP::GeV);
+	  theInelasticProcess->RegisterMe(theBertiniModel);
+
+	  // Added FTFP Model - SS
+	  FTFP_model->SetMinEnergy(4.0*CLHEP::GeV);
+	  theInelasticProcess->RegisterMe(FTFP_model);
 	  pmanager->AddDiscreteProcess(theInelasticProcess);
```

Comment:
It looks like there is an overlap in energy ranges for the Bertini and FTFP models in the pi+ particles between 4 and 5 GeV. @fdilodov, I see you put in the Bertini part. Do you know anything about this? If the overlap is a problem, which model should be adjusted?